mergify_engine/web/api/queues.py

```diff
@@ -317,5 +325,367 @@ async def repository_queues(
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    queue_name: rules.QueueName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the queue"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+) -> responses.JSONResponse:
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token as access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+        config_file = await repository_ctxt.get_mergify_config_file()
+
+        if config_file is not None:
+            config = get_mergify_config(config_file)
+            queue_rules = config["queue_rules"]
+            if all(queue_name != rule.name for rule in queue_rules):
+                return responses.JSONResponse(
+                    status_code=400,
+                    content={"error": f'The queue "{queue_name}" does not exists'},
+                )
+
+            qf = await QueueFreeze.get(repository_ctxt, queue_name)
+            if qf is not None:
+                return responses.JSONResponse(
+                    status_code=200,
+                    content={
+                        "queue_freeze_data": qf.get_json(),
+                        "description": f'The queue "{queue_name}" is currently frozen',
+                    },
+                )
+            else:
+                return responses.JSONResponse(
+                    status_code=200,
+                    content={
+                        "queue_freeze_data": False,
+                        "description": f'The queue "{queue_name}" is not currently frozen',
+                    },
+                )
+        else:
+            return responses.JSONResponse(
+                status_code=400,
+                content={
+                    "error": "Configuration file .mergify.yml is missing",
+                },
+            )
+
+
+@router.get(
+    "/repos/{owner}/{repository}/queues/freeze",  # noqa: FS003
```

Comment:
```suggestion
    "/repos/{owner}/{repository}/queues/freezes",  # noqa: FS003
```
?
Since it returns a list?